### ISSUE TITLE
Support GRPCs protocol (supercedes #899)

### ISSUE THREAD

**Issue by ghostsquad** [kubernetes-sigs/aws-load-balancer-controller#1699]
supercedes #899 

https://aws.amazon.com/about-aws/whats-new/2020/10/application-load-balancers-enable-grpc-workloads-end-to-end-http-2-support/
https://aws.amazon.com/blogs/aws/new-application-load-balancer-support-for-end-to-end-http-2-and-grpc/

**Comment by ghostsquad** [comment]
I believe GRPC is actually already supported, but it doesn't appear in the documentation. I'll test and report back.

https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/a08ca0d9ad49cf7ffb4234c9ee466c9517944521/pkg/annotations/constants.go?ts=4#L27

https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/f95827224a66cae20a1af999d8ef1d46ec462547/pkg/ingress/model_build_target_group.go?ts=4#L217

**Comment by ghostsquad** [comment]
this appears to have created an ingress successfully:

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    alb.ingress.kubernetes.io/certificate-arn: <REDACTED>
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}, {"HTTPS":443}]'
    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}'
    alb.ingress.kubernetes.io/scheme: internal
    alb.ingress.kubernetes.io/ssl-passthrough: "true"
    alb.ingress.kubernetes.io/target-type: ip
    alb.ingress.kubernetes.io/backend-protocol-version: GRPC
    alb.ingress.kubernetes.io/healthcheck-protocol-version: GRPC
    kubernetes.io/ingress.class: alb
  name: argocd-server-grpc
  namespace: argocd
spec:
  rules:
  - http:
      paths:
      - backend:
          serviceName: ssl-redirect
          servicePort: use-annotation
        path: /*
      - backend:
          serviceName: argocd-server
          servicePort: 8080
        path: /*
```

However, it appears the Target Groups don't reflect the protocol version as GRPC

**Comment by M00nF1sh** [comment]
@ghostsquad 
Hi, the GRPC support will be released as part of our v2.1.0 release soon. stay tuned.

**Comment by M00nF1sh** [comment]
closing this as v2.1.0 is released :D
